modules/admanBidAdapter.js

```diff
@@ -0,0 +1,79 @@
+import {registerBidder} from '../src/adapters/bidderFactory';
+import * as utils from '../src/utils';
+
+const BIDDER_CODE = 'adman';
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: ['video', 'banner'],
+  isBidRequestValid: function(bid) {
+    let isValid = false;
+    if (typeof bid.params !== 'undefined') {
+      let isValidId = _validateId(utils.getValue(bid.params, 'id'));
+      isValid = isValidId;
+    }
+
+    if (!isValid) {
+      utils.logError('Adman id parameter is required. Bid aborted.');
+    }
+    return isValid;
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    const ENDPOINT_URL = '//bidtor.admanmedia.com/prebid';
+    const bids = validBidRequests.map(buildRequestObject);
+    const payload = {
+      referrer: utils.getTopWindowUrl(),
+      bids,
+      deviceWidth: screen.width
+    };
+
+    if (bidderRequest && bidderRequest.gdprConsent) {
+      payload.gdpr = {};
+      payload.gdpr.consent = bidderRequest.gdprConsent.consentString;
+      payload.gdpr.applies = bidderRequest.gdprConsent.gdprApplies;
+    } else {
+      payload.gdpr.consent = '';
```

Comment:
The assignment of `payload.gdpr.consent` to empty string is being overwritten by `null` on the next line. Either delete **line 35** (if this is working correctly server-side), or change it to a object literal like this:
```
payload.gdpr = {
  consent: ''
}
```